app/src/ui/publish-repository/publish.tsx

```diff
@@ -237,27 +302,56 @@ export class Publish extends React.Component<IPublishProps, IPublishState> {
         settings.description,
         settings.private,
         account,
-        settings.org
+        org
       )
 
       this.props.onDismissed()
     } catch (e) {
-      this.setState({ error: e, publishing: false })
+      this.setCurrentTabError(e)
+      this.setState({ publishing: false })
     }
   }
 
   private onTabClicked = (index: PublishTab) => {
     const isTabChanging = index !== this.state.currentTab
     if (isTabChanging) {
-      // Clear the selected org since dot com and Enterprise will have a different
-      // set of orgs.
-      const settings = { ...this.state.publishSettings, org: null }
-      this.setState({ currentTab: index, publishSettings: settings })
-      // Swap the current stored error from the active tab with the error from
-      // the inactive tab. So that each tab saves and displays their own error.
-      const temporaryError: Error | null = this.state.error
-      this.setState({ error: this.state.inactiveTabError })
-      this.setState({ inactiveTabError: temporaryError })
+      this.setState({ currentTab: index })
+    }
+  }
+
+  private getCurrentTabState = () =>
+    this.state.currentTab === PublishTab.DotCom
+      ? this.state.dotcomTabState
+      : this.state.gheTabState
+
+  private setTabState = (state: TabState) => {
+    if (state.kind === 'enterprise') {
+      this.setState({ gheTabState: { ...state } })
```

Comment:
There's a few places where we unnecessarily spread:

```suggestion
      this.setState({ gheTabState: state })
```